### ISSUE TITLE
Calendar Block Color options are not working properly

### ISSUE THREAD

**Issue by pitamdey** [WordPress/gutenberg#61346]
### Description

In Calendar Block the colour options are not working properly, like when we change the colour of the text or background it is not reflected in the editor side unless we refresh the whole page. So it should not happen the text & background color should reflect in editor also when we change it at the same time like another block

### Step-by-step reproduction instructions

1. Add Calendar Block
2. Change the text colour & background colour
3. Now you can not see any changes on the editor side but when we preview we can see the changes in frontend
4. On Refresh the page now we can see the changes in the editor

### Screenshots, screen recording, code snippet

Recording : https://www.loom.com/share/8afd314830fd499bbbfa12386a7e0540?sid=541243c7-4965-4b8f-9598-663902bd7021

### Please confirm that you have searched existing issues in the repo.

Yes

### Please confirm that you have tested with all plugins deactivated except Gutenberg.

Yes

**Comment by t-hamano** [comment]
Thanks for the report. I was also able to reproduce this problem.

When I went back through the commit history, I found that this problem occurred in #56875.

Strangely, the settings in the Typography panel take effect immediately, but the settings in the Color panel don't take effect until I save the post. Maybe it has something to do with the calendar block skipping serialization and using the ServerSideRender component.

cc @ellatrix 

https://github.com/WordPress/gutenberg/assets/54422211/3ced4ef7-48f5-45d9-8ba1-2f75f7537df7

**Comment by Rishit30G** [comment]
> Strangely, the settings in the Typography panel take effect immediately,

Yes it does take effect, but once we save the post and come back to editor to change the font size, we face similar issue just like the text and background colour 

https://github.com/user-attachments/assets/0f17571b-e60e-45a0-8663-8c899aa3786d

EDIT: Similar Comment https://github.com/WordPress/gutenberg/pull/64986#issuecomment-2348246240

**Comment by Rishit30G** [comment]
Related https://github.com/WordPress/gutenberg/issues/64739

**Comment by t-hamano** [comment]
I have abstracted this problem and submitted it as #69220. Therefore, let's close this issue.
